--- kernel-mizy.ro/drivers/i2c/busses/i2c-sunxi.c	2017-01-24 00:38:28.000000000 +0700
+++ kernel-mizy.rw/drivers/i2c/busses/i2c-sunxi.c	2017-03-23 21:56:11.395359267 +0700
@@ -982,7 +982,8 @@ static int sunxi_i2c_do_xfer(struct sunx
 		ret = -ETIME;
 	}
 	else if (ret != num) {
-		I2C_ERR("[i2c%d] incomplete xfer (status: 0x%x, dev addr: 0x%x)\n", i2c->bus_num, ret, msgs->addr);
+//		i2c scan make alot msg like this 
+//		I2C_ERR("[i2c%d] incomplete xfer (status: 0x%x, dev addr: 0x%x)\n", i2c->bus_num, ret, msgs->addr);
 		ret = -ECOMM;
 	}
 out:
