
## hyphop ##

--- kernel-mizy.ro/drivers/mmc/host/Makefile	2017-01-24 00:38:32.000000000 +0700
+++ kernel-mizy.rw/drivers/mmc/host/Makefile	2017-05-31 10:48:18.613642945 +0700
@@ -24,7 +24,7 @@ obj-$(CONFIG_MMC_TIFM_SD)	+= tifm_sd.o
 obj-$(CONFIG_MMC_MSM)		+= msm_sdcc.o
 obj-$(CONFIG_MMC_MVSDIO)	+= mvsdio.o
 obj-$(CONFIG_MMC_DAVINCI)       += davinci_mmc.o
-obj-$(CONFIG_MMC_SPI)		+= mmc_spi.o
+obj-m				+= mmc_spi.o
 ifeq ($(CONFIG_OF),y)
 obj-$(CONFIG_MMC_SPI)		+= of_mmc_spi.o
 endif
